overlord/devicestate/devicemgr.go

```diff
@@ -107,16 +132,39 @@ func (m *DeviceManager) ensureOperational() error {
 		return nil
 	}
 
+	gadgetInfo, err := snapstate.GadgetInfo(m.state)
+	if err == state.ErrNoState {
+		// no gadget installed yet, cannot proceed
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
 	// XXX: some of these will need to be split and use hooks
 	// retries might need to embrace more than one "task" then,
 	// need to be careful
 
+	tasks := []*state.Task{}
+
+	var deviceInit *state.Task
+	if gadgetInfo.Hooks["device-init"] != nil {
+		summary := i18n.G("Run device init hook")
```

Comment:
Having the real name here (device-initializing for now) will likely be nicer for passers-by to find their way.